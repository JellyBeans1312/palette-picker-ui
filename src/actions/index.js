export const addProject = id => ({
  type: 'ADD_PROJECT',
  id
});

export const addAllProjects = allProjects => ({
  type: 'ADD_ALL_PROJECTS',
  allProjects
});

export const saveColors = colors => ({
  type: 'SAVE_COLORS',
  colors
});