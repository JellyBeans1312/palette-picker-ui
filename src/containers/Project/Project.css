.project-header {
  display: flex;
}

.small-palettes {
  display:flex;
}

.small-palettes-ul {
  display: flex;
  list-style-type: none;
  padding: 0;
}