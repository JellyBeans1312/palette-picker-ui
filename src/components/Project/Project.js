import React from 'react';
import './Project.css';
import { addPalette } from '../../actions';
import { connect } from 'react-redux';

const Project = ({props, palettes, addPalette}) => {
  const displayPalettes = palettes.map(palette => {
    const { updated_at, created_at, palette_name, project_id, id, ...newPalette } = palette
    let keys = Object.keys(newPalette);
    return  (
      <ul onClick={() => addPalette(newPalette)}>
        <li>{palette.palette_name}
          { keys.map(key => {
            return <div style={{ backgroundColor: newPalette[key], height: 20, width: 20 }}></div>
          })}
        </li>
      </ul>
    )
  })

  const { project_name } = props    
  return (
    <div>
      {project_name}
      {displayPalettes}
    </div>
  )
}

export const mapDispatchToProps = dispatch => ({
  addPalette: palette => dispatch(addPalette(palette))
})

export default connect(null, mapDispatchToProps)(Project);