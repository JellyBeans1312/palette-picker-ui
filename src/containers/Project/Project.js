import React from 'react';
import './Project.css';
import { addPalette, removePalette, removeProject, addProject } from '../../actions';
import { connect } from 'react-redux';
import trashcan from '../../assets/trashcan.svg';
import editIcon from '../../assets/editIcon.svg'
import { deletePalette, deleteProject } from '../../util/apiCalls'

const Project = ({props, palettes, addPalette, removePalette, removeProject, addProject}) => {
  const displayPalettes = palettes.map(palette => {
    const { updated_at, created_at, palette_name, project_id, ...newPalette } = palette
    const { id, ...displayPalette } = newPalette
    let keys = Object.keys(displayPalette);
    return  (
      <div className='small-palettes-container'>
        <ul className='small-palettes-ul'>
          <li className='small-palettes'>
            <div>{palette.palette_name}</div>
            { keys.map(key => {          
              return <div className='palette-colors' style={{ backgroundColor: newPalette[key]}}></div>
            })}
          </li>
        </ul>
        <img src={trashcan} alt='trashcan' className='palette-buttons' onClick={() => {deletePalette(palette.id); removePalette(palette.id)}}></img>
        <img src={editIcon} alt='edit' className='palette-buttons' onClick={() => addPalette(newPalette)}/>

      </div>
    )
  })

  const { project_name, id } = props    

  return (
    <div>
      <div className='project-header'>
        <div>
          {project_name}
          <button onClick={() => addProject(project_name, id)}>Select this project</button>
        </div>
      <img src={trashcan} className='project-button' style={{ height: 30, width: 30}} onClick={() => {deleteProject(id); removeProject(id)}}></img>
      </div>
      {displayPalettes}
    </div>
  )
}

export const mapDispatchToProps = dispatch => ({
  addPalette: palette => dispatch(addPalette(palette)),
  removePalette: id => dispatch(removePalette(id)),
  removeProject: id => dispatch(removeProject(id)),
  addProject: (project_name, id) => dispatch(addProject(project_name, id))
});

export default connect(null, mapDispatchToProps)(Project);