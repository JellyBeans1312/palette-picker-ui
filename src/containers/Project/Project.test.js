import React from 'react';
import { shallow } from 'enzyme';
import { Project, mapDispatchToProps } from './Project';
import { addPalette, removePalette, removeProject, addProject } from '../../actions';

describe('Project', () => {
  it('should match snapshot', () => {
    const mockPalettes = [
      {
      id: 22,
      palette_name: "asdf",
      project_id: 50,
      color_one: "#67cd51",
      color_two: "#15fe0d",
      color_three: "#507dbb",
      color_four: "#784994",
      color_five: "#2570dd"
    },
    {
      id: 23,
      palette_name: "fdsa",
      project_id: 50,
      color_one: "#67cd51",
      color_two: "#15fe0d",
      color_three: "#507dbb",
      color_four: "#784994",
      color_five: "#2570dd"
    },
    {
      id: 21,
      palette_name: "afsd",
      project_id: 50,
      color_one: "#67cd51",
      color_two: "#15fe0d",
      color_three: "#507dbb",
      color_four: "#784994",
      color_five: "#2570dd"
    }
  ]
    const wrapper = shallow(
    <Project
      props={{
        id: 1,
        project_name: 'pants',
        created_at: '2',
        updated_at: '3'
      }}
      palettes={mockPalettes}
      addPalette={jest.fn()}
      removePalette={jest.fn()}
      removeProject={jest.fn()}
      addProject={jest.fn()}
    />)

    expect(wrapper).toMatchSnapshot()
  });
  describe('mapDispatchToProps', () => {
    it('should call action addPalette', () => {
      const mockPalette = {
        id: 1,
        palette_name: 'something'
      };
      const mockDispatch = jest.fn();
      const actionToDispatch = addPalette(mockPalette);
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addPalette(mockPalette);
  
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
    it('should call action removePalette', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = removePalette(1);
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.removePalette(1);
  
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
    it('should call action addProject', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = addProject('pants', 1);
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addProject('pants', 1);
  
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
    it('should call action removeProject', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = removeProject(1);
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.removeProject(1);
  
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  })
})