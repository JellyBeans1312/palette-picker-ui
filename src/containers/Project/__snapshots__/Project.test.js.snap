// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should match snapshot 1`] = `
<div>
  <div
    className="project-header"
  >
    <div>
      pants
      <button
        onClick={[Function]}
      >
        Select this project
      </button>
    </div>
    <img
      alt="trashcan"
      className="project-button"
      onClick={[Function]}
      src="trashcan.svg"
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
  </div>
  <div
    className="small-palettes-container"
  >
    <ul
      className="small-palettes-ul"
    >
      <li
        className="small-palettes"
      >
        <div>
          asdf
        </div>
        <div
          className="palette-colors"
          style={
            Object {
              "backgroundColor": "#67cd51",
            }
          }
        />
        <div
          className="palette-colors"
          style={
            Object {
              "backgroundColor": "#15fe0d",
            }
          }
        />
        <div
          className="palette-colors"
          style={
            Object {
              "backgroundColor": "#507dbb",
            }
          }
        />
        <div
          className="palette-colors"
          style={
            Object {
              "backgroundColor": "#784994",
            }
          }
        />
        <div
          className="palette-colors"
          style={
            Object {
              "backgroundColor": "#2570dd",
            }
          }
        />
      </li>
    </ul>
    <img
      alt="trashcan"
      className="palette-buttons"
      onClick={[Function]}
      src="trashcan.svg"
    />
    <img
      alt="edit"
      className="palette-buttons"
      onClick={[Function]}
      src="editIcon.svg"
    />
  </div>
  <div
    className="small-palettes-container"
  >
    <ul
      className="small-palettes-ul"
    >
      <li
        className="small-palettes"
      >
        <div>
          fdsa
        </div>
        <div
          className="palette-colors"
          style={
            Object {
              "backgroundColor": "#67cd51",
            }
          }
        />
        <div
          className="palette-colors"
          style={
            Object {
              "backgroundColor": "#15fe0d",
            }
          }
        />
        <div
          className="palette-colors"
          style={
            Object {
              "backgroundColor": "#507dbb",
            }
          }
        />
        <div
          className="palette-colors"
          style={
            Object {
              "backgroundColor": "#784994",
            }
          }
        />
        <div
          className="palette-colors"
          style={
            Object {
              "backgroundColor": "#2570dd",
            }
          }
        />
      </li>
    </ul>
    <img
      alt="trashcan"
      className="palette-buttons"
      onClick={[Function]}
      src="trashcan.svg"
    />
    <img
      alt="edit"
      className="palette-buttons"
      onClick={[Function]}
      src="editIcon.svg"
    />
  </div>
  <div
    className="small-palettes-container"
  >
    <ul
      className="small-palettes-ul"
    >
      <li
        className="small-palettes"
      >
        <div>
          afsd
        </div>
        <div
          className="palette-colors"
          style={
            Object {
              "backgroundColor": "#67cd51",
            }
          }
        />
        <div
          className="palette-colors"
          style={
            Object {
              "backgroundColor": "#15fe0d",
            }
          }
        />
        <div
          className="palette-colors"
          style={
            Object {
              "backgroundColor": "#507dbb",
            }
          }
        />
        <div
          className="palette-colors"
          style={
            Object {
              "backgroundColor": "#784994",
            }
          }
        />
        <div
          className="palette-colors"
          style={
            Object {
              "backgroundColor": "#2570dd",
            }
          }
        />
      </li>
    </ul>
    <img
      alt="trashcan"
      className="palette-buttons"
      onClick={[Function]}
      src="trashcan.svg"
    />
    <img
      alt="edit"
      className="palette-buttons"
      onClick={[Function]}
      src="editIcon.svg"
    />
  </div>
</div>
`;
